core/gather/gatherers/seo/tap-targets.js

```diff
@@ -192,94 +192,103 @@ function disableFixedAndStickyElementPointerEvents(className) {
+    // is at the current scroll position)
+    const reenableFixedAndStickyElementPointerEvents =
+      disableFixedAndStickyElementPointerEvents(className);
+
+    /** @type {{
+      tapTargetElement: Element,
+      visibleClientRects: LH.Artifacts.Rect[]
+    }[]} */
+    const tapTargetsWithVisibleClientRects = [];
+    // We use separate loop here to get visible client rects because that involves
+    // scrolling around the page for elementCenterIsAtZAxisTop, which would affect the
+    // client rect positions.
+    tapTargetsWithClientRects.forEach(({tapTargetElement, clientRects}) => {
+      // Filter out empty client rects
+      let visibleClientRects = clientRects.filter(cr => cr.width !== 0 && cr.height !== 0);
+
+      // Filter out client rects that are invisible, e.g because they are in a position absolute element
+      // with a lower z-index than the main content.
+      // This will also filter out all position fixed or sticky tap targets elements because we disable pointer
+      // events on them before running this. That's the correct behavior because whether a position fixed/stick
+      // element overlaps with another tap target depends on the scroll position.
+      visibleClientRects = visibleClientRects.filter(rect => {
+        // Just checking the center can cause false failures for large partially hidden tap targets,
+        // but that should be a rare edge case
+        // @ts-expect-error - put into scope via stringification
+        const rectCenterPoint = getRectCenterPoint(rect);
+        return elementCenterIsAtZAxisTop(tapTargetElement, rectCenterPoint);
+      });
+
+      if (visibleClientRects.length > 0) {
+        tapTargetsWithVisibleClientRects.push({
+          tapTargetElement,
+          visibleClientRects,
+        });
+      }
     });
 
-    if (visibleClientRects.length > 0) {
-      tapTargetsWithVisibleClientRects.push({
-        tapTargetElement,
-        visibleClientRects,
+    for (const {tapTargetElement, visibleClientRects} of tapTargetsWithVisibleClientRects) {
+      targets.push({
+        clientRects: visibleClientRects,
+        href: /** @type {HTMLAnchorElement} */(tapTargetElement)['href'] || '',
+        // @ts-expect-error - getNodeDetails put into scope via stringification
+        node: getNodeDetails(tapTargetElement),
       });
     }
-  });
 
-  for (const {tapTargetElement, visibleClientRects} of tapTargetsWithVisibleClientRects) {
-    targets.push({
-      clientRects: visibleClientRects,
-      href: /** @type {HTMLAnchorElement} */(tapTargetElement)['href'] || '',
-      // @ts-expect-error - getNodeDetails put into scope via stringification
-      node: getNodeDetails(tapTargetElement),
-    });
-  }
-
-  reenableFixedAndStickyElementPointerEvents();
+    reenableFixedAndStickyElementPointerEvents();
 
-  return targets;
+    return targets;
+  } finally {
+    window.scrollTo(originalScrollPosition.x, originalScrollPosition.y);
+  }
```

Comment:
To preserve git blame a bit, and make this change easy to un-do if needed, and to make it easier to read: I think it'd better to have a wrapper function that handles remembering the scroll state, running the gatherer impl, then finally restoring the state.